llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -504,6 +504,15 @@ static VPValue *createScalarIVSteps(VPlan &Plan, const InductionDescriptor &ID,
     HeaderVPBB->insert(BaseIV->getDefiningRecipe(), IP);
   }
 
+  VPTypeAnalysis TypeInfo(SE.getContext());
+  if (TypeInfo.inferScalarType(BaseIV) != TypeInfo.inferScalarType(Step)) {
+    Step = new VPScalarCastRecipe(Instruction::Trunc, Step,
```

Comment:
Does this recipe only generate Trunc? No need to hold an Opcode, can be renamed VPScalarTruncRecipe. Assert that the type sizes correspond to Trunc?

Reply:
Yes, here the only use-case is `Trunc`, but the implementation of the recipe is intentionally kept generic for any cast where only the first lane is used, so it can be used directly in #76172

Reply:
Assert that the type sizes correspond to Trunc?